src/main/java/uk/gov/companieshouse/pscstatement/delta/service/ResponseHandler.java

```diff
@@ -4,27 +4,28 @@
 import java.util.Map;
 import org.springframework.stereotype.Service;
 import uk.gov.companieshouse.api.error.ApiErrorResponseException;
-import uk.gov.companieshouse.api.handler.delta.pscstatements.request.PscStatementsDelete;
+import uk.gov.companieshouse.api.handler.Executor;
 import uk.gov.companieshouse.api.handler.exception.URIValidationException;
 import uk.gov.companieshouse.api.model.ApiResponse;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.pscstatement.delta.exception.NonRetryableErrorException;
 import uk.gov.companieshouse.pscstatement.delta.exception.RetryableErrorException;
 
+@SuppressWarnings("unchecked")
 @Service
-public class ResponseHandler {
+public class ResponseHandler<T> {
 
     /**
      * Handle response from data api.
      */
     public ApiResponse<Void> handleApiResponse(Logger logger, String context, String operation,
-                                               String uri, PscStatementsDelete delete) {
+                                               String uri, T executor) {
```

Comment:
Here, can we not specify the executor as "Executor<ApiResponse<Void>> executor"?

Reply:
I don't think so because I think we couldn't use the Executor class directly, it needed to be either the put or delete executor. But in making this piece of code generic to both executors so we didn't have to duplicate it, we had to make it a generic T type.